bus/bus.go

```diff
@@ -1547,16 +1565,40 @@ func (b *bus) Handler() http.Handler {
 		"PUT    /setting/:key": b.settingKeyHandlerPUT,
 		"DELETE /setting/:key": b.settingKeyHandlerDELETE,
 
+		"GET    /state": b.stateHandlerGET,
+
 		"GET    /params/upload":  b.paramsHandlerUploadGET,
 		"GET    /params/gouging": b.paramsHandlerGougingGET,
 	}))
 }
 
+// Run starts the bus.
+func (b *bus) Run() error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	if !b.startTime.IsZero() {
+		return errors.New("bus already running")
+	}
+	b.startTime = time.Now()
```

Comment:
I'm not in love with adding a `Run` method to the bus and returning it in some roundabout way only to get a `startTime` set here.
Why not set this at the end of `New`?